sdk/core/azure-core/test/ut/context.cpp

```diff
@@ -106,28 +115,34 @@ TEST(Context, NestedIsCancelled)
   EXPECT_FALSE(c2.IsCancelled());
   EXPECT_TRUE(c3.IsCancelled());
 
-  EXPECT_TRUE(c2.HasKey(key));
-  EXPECT_FALSE(context.HasKey(key));
-  EXPECT_FALSE(c3.HasKey(key));
+  value = "b";
+  EXPECT_TRUE(c2.TryGetValue<std::string>(key, value));
+  EXPECT_EQ(value, "Value");
+  EXPECT_FALSE(context.TryGetValue<std::string>(key, value));
+  EXPECT_FALSE(c3.TryGetValue<std::string>(key, value));
 }
 
 TEST(Context, CancelWithValue)
 {
   Context context;
   Context::Key const key;
 
-  auto c2 = context.WithValue(key, "Value");
+  std::string actualValue = "Value";
+  auto c2 = context.WithValue(key, actualValue);
   EXPECT_FALSE(context.IsCancelled());
   EXPECT_FALSE(c2.IsCancelled());
-  EXPECT_TRUE(c2.HasKey(key));
-  EXPECT_FALSE(context.HasKey(key));
+  std::string value = "a";
+  EXPECT_TRUE(c2.TryGetValue<std::string>(key, value));
+  EXPECT_EQ(value, "Value");
+  EXPECT_FALSE(context.TryGetValue<std::string>(key, value));
```

Comment:
Verify that the value hasn't changed when `false` is returned

Reply:
Will skip adding it everywhere, like here and below. It doesn't add value to duplicate the testing.